### PR TITLE
Make mypy_wrapper.py accept multiple filenames

### PR DESCRIPTION
**Pull request by samestep**
A followup to #57752.

**Test plan:**

```
mypy --config=mypy-strict.ini
python tools/test/test_mypy_wrapper.py
python tools/test/test_actions_local_runner.py -k mypy
```

### CODE REVIEW

File: tools/actions_local_runner.py
Comment (seemethere) [inline tools/actions_local_runner.py:17]: Should we use this?

```suggestion
from . import mypy_wrapper
```

File: tools/actions_local_runner.py
Comment (driazati) [inline tools/actions_local_runner.py:174]: This should still call it via `shell_cmd` since that spins off a separate process to do the actual computation that we can `await`, whereas I think this will block the runner thread until mypy is done.

File: tools/mypy_wrapper.py
Comment (driazati) [inline tools/mypy_wrapper.py:36]: hope and windows don't usually go together

File: tools/actions_local_runner.py
Comment (samestep) [reply tools/actions_local_runner.py]: I tried that and it didn't work: https://github.com/pytorch/pytorch/runs/2558431639 any other suggestions?

File: tools/mypy_wrapper.py
Comment (samestep) [reply tools/mypy_wrapper.py:67]: Sure, I'll add comments to these functions describing their purpose 👍

> would it work if we turned all the non-specific files in the config into globs (i.e. append a `*`) and then use [fnmatch](https://docs.python.org/3/library/fnmatch.html)?

Maybe? It probably doesn't matter which approach we use since the lists of files are going to be very small, but I chose this trie approach because it has better asymptotic complexity than what you suggest (since this way you only have to do a lookup rather than iterate through everything every time). Again though, asymptotic complexity obviously doesn't matter here haha

File: tools/mypy_wrapper.py
Comment (driazati) [reply tools/mypy_wrapper.py:36]: nah I don't think we need to support Windows with `tools/`

File: tools/actions_local_runner.py
Comment (driazati) [reply tools/actions_local_runner.py:174]: Even as an async job it still all runs in the one Python thread, if Make decides to run the mypy target before any others then they would get blocked by mypy (it's the same as old multi-threading in Python under the GIL vs. `multiprocessing`), so we kind of have to have another interpreter process to handle the actual mypy compute, the runner should just be dispatching jobs and collecting their results, not doing anything intensive.

File: tools/mypy_wrapper.py
Comment (driazati) [reply tools/mypy_wrapper.py:67]: I generally err towards whatever is less code, especially if it's something from the standard library. These tools scripts especially aren't the most documented so people will probably come looking into the code to see what it does at some point, and the simpler it is the better imo

File: tools/actions_local_runner.py
Comment (samestep) [reply tools/actions_local_runner.py:174]: Ok gotcha, I'll make that change (that'll probably also fix the weird `import` issue I've been hitting in the "Test tools" workflow)

**Review by driazati — APPROVED** [review]
lgtm after comments

**Pull request closed without merging**
